File: tests/fixtures/kotlin/Service.kt
// User data class
data class User(val id: Int, val name: String, val email: String)

// Repository interface
interface Repository {
    fun findById(id: Int): User?
    fun save(user: User)
    fun count(): Int
}

// Logger interface
interface Logger {
    fun log(message: String)
}

// UserService class
class UserService(
    private val repo: Repository,
    private val logger: Logger
) {
    fun findById(id: Int): User? {
        logger.log("Finding user $id")
        return repo.findById(id)
    }

    fun save(user: User) = repo.save(user)
    fun count(): Int = repo.count()
}

// AbstractController open class
abstract class AbstractController(protected val viewPath: String) {
    fun render(template: String): String = "$viewPath/$template"
}

// UserController extends AbstractController
class UserController(
    private val userService: UserService
) : AbstractController("users") {
    fun index(): String = render("index")
    fun show(id: Int): String {
        val user = userService.findById(id)
        return render("show")
    }
}

// Utils object
object Utils {
    fun formatName(name: String) = "User: $name"
}

File: tests/fixtures/dart/service.dart
// User class
class User {
  final int id;
  final String name;
  final String email;

  User(this.id, this.name, this.email);
}

// Repository abstract class
abstract class Repository {
  User? findById(int id);
  void save(User user);
  int count();
}

// Logger mixin
mixin Logger {
  void log(String message) {
    print('[LOG] $message');
  }
}

// UserService class with Logger mixin
class UserService with Logger {
  final Repository repo;

  UserService(this.repo);

  User? findById(int id) {
    log('Finding user $id');
    return repo.findById(id);
  }

  void save(User user) {
    repo.save(user);
  }

  int count() => repo.count();
}

// AbstractController
abstract class AbstractController {
  final String viewPath;

  AbstractController(this.viewPath);

  String render(String template) {
    return '$viewPath/$template';
  }
}

// UserController extends AbstractController
class UserController extends AbstractController {
  final UserService userService;

  UserController(this.userService) : super('users');

  String index() => render('index');
  String show(int id) {
    final user = userService.findById(id);
    return render('show');
  }
}

// Utils class (namespace)
class Utils {
  static String formatName(String name) {
    return 'User: $name';
  }
}
